Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1

```diff
@@ -2968,8 +2968,8 @@ function Get-M365DSCExportContentForResource
     if ($partialContent.ToLower().IndexOf($OrganizationName.ToLower()) -gt 0)
     {
         $partialContent = $partialContent -ireplace [regex]::Escape($OrganizationName + ':'), "`$($OrganizationName):"
+        $partialContent = $partialContent -ireplace [regex]::Escape('@' + $OrganizationName), "@`$TenantId"
```

Comment:
In the case of creds based auth, tenantid will always be null. We unfortunately cannot replace this across the board. My recommendation would be to check if the TenantID is populated before attempting to replace.

Reply:
Good point, I have to admit I've been focusing on app-based authentication at the moment. I will take a look at your recommendation.

Reply:
If we look at the part where the generalization takes place we can see something else:
No matter which method is used to authenticate, the searched string replaced by `$OrganizationName is always a complete domain (e.g. "contoso.onmicrosoft.com")
https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1#L2912-L2921

Therefore, the entire domain (contoso.onmicrosoft.com) is always replaced by `$OrganizationName during ireplace. 
But the $OrganizationName variable is only filled with "contoso.onmicrosoft.com" in case of authentication with credentials, see example below, else $OrganizationName contains only "contoso".

https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1#L2968-L2973


Example:

Username: admin@contoso.microsoft.com

When the authentication is app-based, the $OrganizationName is only filled with the first part of the subdomain.
See this code:
https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCReverse.psm1#L264-L267

Result in ConfigurationData.psd1: 
$OrganizationName = "contoso"

But if you authenticate with credentials then this process is not considered at all:
https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCReverse.psm1#L231-L267

and only the trailing part of the email address is used in the following lines.

https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCReverse.psm1#L416

Result in the ConfigarationData.psd1:
$OrganizationName  = "contoso.microsoft.com"

Conclusion:
Judging by the description of $OrganizationName variable and some other functions where also $OrganizationName is determined, I think that the complete domain belongs in this variable. Therefore, in my opinion, the following lines should be removed.
https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCReverse.psm1#L264-L267

Reply:
After removing the lines, an mof file could be compiled successfully with both app-based and credentials-based authentication.
https://github.com/microsoft/Microsoft365DSC/blob/7cd5188842d1250142b954f90f4ef22f1b830aac/Modules/Microsoft365DSC/Modules/M365DSCReverse.psm1#L264-L267

Reply:
The problem could come from commit #[87735be](https://github.com/microsoft/Microsoft365DSC/pull/2387/commits/87735bef29450c8b154e388f91b85e0232a02320) , because there the variable $principal became $organization and thus the variable is overwritten with the "NetBios" name of the tenant.

So instead of removing the lines completely, it would also work to rename the variable back to $principal.